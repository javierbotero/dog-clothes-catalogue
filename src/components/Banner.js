import React from 'react';
import PropTypes from 'prop-types';

const Banner = props => {
  const { clothes, picturesDirectory } = props;
  const item = clothes[Math.floor(Math.random() * clothes.length)];
  const style = {
    backgroundImage: `url(${picturesDirectory[item.image[0]][item.image[1]]})`,
  };
  return (
    <div key={item.id}>
      <div style={style} className="picture-banner" />
      <span>{item.price}</span>
      <span>{item.name}</span>
    </div>
  );
};

Banner.propTypes = {
  clothes: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.arrayOf(PropTypes.oneOfType([
      PropTypes.string.isRequired,
      PropTypes.number.isRequired,
    ])).isRequired,
    id: PropTypes.number.isRequired,
  })).isRequired,
  picturesDirectory: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
};

export default Banner;
