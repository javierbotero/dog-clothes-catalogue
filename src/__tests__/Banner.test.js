import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'regenerator-runtime/runtime';
import '@testing-library/jest-dom';
import Banner from '../components/Banner';
import renderer from 'react-test-renderer';
import {
  render,
  PICTURES_DIRECTORY,
  clothes,
  screen,
} from './test-util/test-util';

describe('Tests banner', () => {
  it('Makes sure print the price of an item', () => {
    render(<Banner picturesDirectory={PICTURES_DIRECTORY} clothes={clothes} />);
    expect(screen.getAllByText(/\d*/i).length > 0).toBe(true);
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Banner
          clothes={clothes}
          picturesDirectory={PICTURES_DIRECTORY}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
