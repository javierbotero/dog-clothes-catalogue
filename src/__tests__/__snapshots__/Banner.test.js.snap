// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests banner renders correctly 1`] = `
<div>
  <div
    className="picture-banner"
    style={
      Object {
        "backgroundImage": "url(coat1)",
      }
    }
  />
  <span>
    100
  </span>
  <br />
  <span>
    product1
  </span>
  <br />
</div>
`;
